pulp_container/app/viewsets.py

```diff
@@ -114,7 +116,50 @@ class Meta:
         }
 
 
-class TagViewSet(ReadOnlyContentViewSet):
+Repo = namedtuple("Repo", "type perm")
+mirror_repo = Repo("container.container", "container.view_containerrepository")
+push_repo = Repo("container.container-push", "container.view_containerpushrepository")
+
+
+class ContainerQuerySetViewSet(viewsets.GenericViewSet):
+    def get_queryset(self):
+        qs = super().get_queryset()
+        has_model_push_repo = self.request.user.has_perm(push_repo.perm)
+        has_model_repo = self.request.user.has_perm(mirror_repo.perm)
+        if has_model_push_repo and has_model_repo:
+            return qs
```

Comment:
This would allow to see content that is not associated to any repo.

Reply:
That is a good observation. I can remove this if we want to display to the user only the content present in the repos he can access. He won't be able to see any 'orphaned' content'. This PR does not solve the 'orphaned content' question. If I leave this then any authed user will be able to see orphaned content.

What would be the preference?

Reply:
I don't have a preference either way. Maybe never show orphaned content?

Reply:
I think we can leave it as is and change when we resolve orphaned content question.
IMO, content would need to have its own perms to provide enough flexibility and cover many edge cases.

Reply:
I think we can leave it as is and change when we resolve orphaned content question.
IMO, content would need to have its own perms to provide enough flexibility and cover many edge cases.